resources/assets/js/src/components/chart/Chart.tsx

```diff
@@ -158,37 +150,45 @@ export default class Chart extends React.Component<IProps, IState> {
     }
 
     private loadNewDataToChart(direction: string = null): void {
-        let dateFrom;
-        let dateTo;
         let diff = 0;
 
         if (this.state.dateRange.dateFrom !== null && this.state.dateRange.dateTo !== null) {
             diff = moment(this.state.dateRange.dateTo)
                 .diff(this.state.dateRange.dateFrom) / 1000 / 60;
         } else {
-            diff = 30;
+            diff = INITIAL_DURATION;
         }
 
+        const dates = this.calculateDiffBetweenDates(direction, diff);
+        const dateFrom = DateUtil.formatDateByFormat(dates.dateFrom, this.state.dbDateFormat);
+        const dateTo = DateUtil.formatDateByFormat(dates.dateTo, this.state.dbDateFormat);
+
+        this.loadData(dateFrom, dateTo, direction);
+    }
+
+    private calculateDiffBetweenDates(direction: string, diff: number): {dateFrom: string, dateTo: string} {
+        const firstDate = this.state.dataMeta.firstDate;
+        const lastDate = this.state.dataMeta.lastDate;
+        let dateFrom;
+        let dateTo;
+
         switch (direction) {
             case "minus":
-                dateTo = moment(this.state.dataMeta.firstDate)
-                    .format(this.state.dbDateFormat);
-                dateFrom = moment(this.state.dataMeta.firstDate)
-                    .subtract(diff, "minutes")
-                    .format(this.state.dbDateFormat);
+                dateTo = moment(firstDate);
+                dateFrom = moment(firstDate).subtract(diff, "minutes");
                 break;
             case "plus":
-                dateFrom = moment(this.state.dataMeta.lastDate)
-                    .format(this.state.dbDateFormat);
-                dateTo = moment(this.state.dataMeta.lastDate)
-                    .add(diff, "minutes")
-                    .format(this.state.dbDateFormat);
+                dateFrom = moment(lastDate);
+                dateTo = moment(lastDate).add(diff, "minutes");
                 break;
             default:
                 return null;
         }
 
-        this.loadData(dateFrom, dateTo, direction);
+        return {
+            dateFrom,
+            dateTo,
+        };
     }
 
     private loadInitialData(): void {
```

Comment:
rename to initialize, this does not load initial data, it inits whole component

Reply:
but that function doesn't initialize whole component, it's out of react lifecycle, it loads first data to chart

Reply:
might be also named as initializeCompoments

Reply:
initializeComponentData is ok? that is name that makes sense to me